src/SQLite.Net/TableMapping.cs

```diff
@@ -184,51 +235,76 @@ public Column(PropertyInfo prop, CreateFlags createFlags = CreateFlags.None, ICo
             /// <param name="val"></param>
             public void SetValue(object obj, object val)
             {
-                var propType = _prop.PropertyType;
+                var propType = _infoProvider.GetMemberType(_prop);
                 var typeInfo = propType.GetTypeInfo();
+                object valueToSet;
 
-                if (typeInfo.IsGenericType && propType.GetGenericTypeDefinition() == typeof (Nullable<>))
+                if (typeInfo.IsGenericType && propType.GetGenericTypeDefinition() == typeof(Nullable<>))
                 {
                     var typeCol = propType.GetTypeInfo().GenericTypeArguments;
-                    if (typeCol.Length > 0)
+                    var nullableType = typeCol[0];
+                    var baseType = nullableType.GetTypeInfo().BaseType;
+                    if (baseType == typeof(Enum))
+                    {
+                        valueToSet = AsEnumValue(obj, nullableType, val);
+                    }
+                    else
                     {
-                        var nullableType = typeCol[0];
-                        var baseType = nullableType.GetTypeInfo().BaseType;
-                        if (baseType == typeof (Enum))
-                        {
-                            SetEnumValue(obj, nullableType, val);
-                        }
-                        else
-                        {
-                            _prop.SetValue(obj, val, null);
-                        }
+                        valueToSet = val;
                     }
                 }
-                else if (typeInfo.BaseType == typeof (Enum))
+                else if (typeInfo.BaseType == typeof(Enum))
                 {
-                    SetEnumValue(obj, propType, val);
+                    valueToSet = AsEnumValue(obj, propType, val);
                 }
                 else
                 {
-                    _prop.SetValue(obj, val, null);
+                    valueToSet = val;
                 }
-            }
 
-            private void SetEnumValue(object obj, Type type, object value)
-            {
-                var result = value;
-                if (result != null)
+                // If we're a value tuple then we need to recreate the tuple with the new value and set that
+                // on the property.
+                if (TupleElement != -1)
+                {
+                    var tupleObj = (ITuple)_infoProvider.GetValue(_prop, obj);
+                    var args = new object[tupleObj.Length];
+                    for (var i = 0; i < tupleObj.Length; ++i)
+                    {
+                        args[i] = i == TupleElement ? valueToSet : tupleObj[i];
+                    }
```

Comment:
When/How does this `SetValue` function get called? In many cases i'd expect we want to set the whole tuple at once as a single operation

Reply:
This gets called for each column in the result set. Since a tuple decomposes into multiple columns then this will get called N times, where N is the size of the tuple.

Ultimately we want to solve this with an interface and code generation, but that is not part of this changelist.